### PR TITLE
Fix unshieldedtreasury deposits, add tests, improve doc

### PR DESCRIPTION
**Pull request by andrew-fleming**
**Fix**: `UnshieldedTreasury._deposit` reverted on every call. Its overflow guard passed `UINT128_MAX - amount (≈2¹²⁸)` to `unshieldedBalanceLte`, which fails at runtime with failed to decode for built-in type u64 despite the built-in's declared `Uint<128>` parameter. Both guards now check the module's own `_balances` with in-circuit arithmetic instead of the protocol balance

That also fixes two latent bugs: the protocol balance is a start-of-execution snapshot, so it couldn't see a credit or debit made earlier in the same transaction. It rejected a legitimate receive-then-spend, and let an intra-transaction overspend through to fail as an opaque range fault instead of the named assert

**Add**: test and sim (dry and live). Covers protocol claims in the module's docs, including a regression guard that pins the `unshieldedBalanceLte` operand limit so the fix can't be quietly reverted

**Docs**: warnings for unspendable recipients (accepted, funds provably removed, not a sanctioned burn) and self-sends (zeroes the tracked balance while funds stay spendable), plus a note that zero-amount ops are value-preserving no-ops

## Summary by CodeRabbit

* **Documentation**
  * Clarified treasury balance accounting, transaction behavior, and deposit/send rules.
  * Documented zero-value operations, overflow and insufficiency checks, recipient considerations, and self-send behavior.

* **Tests**
  * Added comprehensive coverage for deposits, sends, multi-call transactions, zero amounts, overflow handling, raw transfers, balance synchronization, and edge cases.
  * Added simulation support for validating treasury operations and balance behavior.

### CODE REVIEW

File: contracts/src/multisig/test/mocks/MockUnshieldedTreasury.compact [hunk 1]
```diff
@@ -26,3 +26,99 @@ export circuit _send(
...
+ * contract is only valid if that contract claims it in the same transaction;
+ * `_send` does not claim, so this is expected to be rejected on-chain.
+ */
+export circuit sendToSelf(color: Bytes<32>, amount: Uint<128>): [] {
+  return Treasury__send(
+    left<ContractAddress, UserAddress>(kernel.self()),
+    color,
+    amount
+  );
+}
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply contracts/src/multisig/test/mocks/MockUnshieldedTreasury.compact:110]: Stale doc. FIxed: [ab6febb](https://github.com/OpenZeppelin/compact-contracts/pull/762/commits/ab6febb75cac7bdaf73a799a26496abf82876887)

**Review by pepebndc — APPROVED** [review]
